packages/sources/aleno/src/transport/price-socketio.ts

```diff
@@ -0,0 +1,125 @@
+import { io, Socket } from 'socket.io-client'
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger, sleep } from '@chainlink/external-adapter-framework/util'
+import {
+  StreamingTransport,
+  SubscriptionDeltas,
+} from '@chainlink/external-adapter-framework/transports/abstract/streaming'
+import { TypeFromDefinition } from '@chainlink/external-adapter-framework/validation/input-params'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { BaseEndpointTypes } from '../endpoint/price'
+import { config } from '../config'
+import { ResponseSchema } from './price-http'
+
+const logger = makeLogger('SocketIOTransport')
+
+export type SocketIOTransportTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: any
+  }
+}
+
+export class SocketIOTransport extends StreamingTransport<SocketIOTransportTypes> {
+  socket?: Socket = undefined
+
+  constructor() {
+    super()
+  }
+
+  override async initialize(
+    dependencies: TransportDependencies<SocketIOTransportTypes>,
+    settings: typeof config.settings,
+    endpointName: string,
+    name: string,
+  ): Promise<void> {
+    super.initialize(dependencies, settings, endpointName, name)
+  }
+
+  establishWsConnection(adaptersettings: typeof config.settings): Socket {
```

Comment:
I think it's probably the right decision to call this in `streamHandler`, but was sticking this in `initialize` considered at any point and why was it not chosen? reconnect? lazy load? something else?

Reply:
In EA-framework both ws and ssetransport initialise their connections in streamHandler.
Followed the same model to maintain uniformity across the code.